components/skus/renderer/skus_js_handler.cc

```diff
@@ -25,6 +27,32 @@
 
 namespace skus {
 
+namespace {
+void ExecuteScript(blink::WebLocalFrame* web_frame, const std::string script) {
+  if (web_frame->IsProvisional())
+    return;
+
+  web_frame->ExecuteScript(
+      blink::WebScriptSource(blink::WebString::FromUTF8(script)));
+}
+void SetProviderNonWritable(blink::WebLocalFrame* web_frame,
+                            const std::string& provider) {
+  const char* provider_str = provider.c_str();
+  const std::string script = base::StringPrintf(
+      R"(;(function() {
+           Object.defineProperty(window.chrome, '%s', {
+             value: window.chrome.%s,
+             configurable: false,
+             writable: false
```

Comment:
what is the goal of setting writable/configurable to be false? because it can be bypassed by simply overriding the `window.chrome` object like this:
<img width="631" alt="Screen Shot 2022-06-09 at 10 48 04 AM" src="https://user-images.githubusercontent.com/549654/172911753-a290faf4-9369-45ed-bca2-6eb5ac14cc07.png">

Reply:
you probably have to set the entire `window.chrome` object to be non-writable and non-configurable like we do for ethereum/solana here https://github.com/brave/brave-core/pull/13240/files#diff-cb2d59317673cd0009f694fde302f571f09871ca9b41bea27699a481fd2000fbR104-R107

Reply:
also i think this only works if this injected code runs before any page-controlled code, otherwise the page can just redefine `Object.defineProperty` to be something else